apps/network-alerts/js/index.js

```diff
@@ -31,23 +47,25 @@
       return;
     }
 
-    var req = navigator.mozSettings.createLock().get(CMAS_ENABLED_KEY);
+    var sendAlertPromise =
+      isPresidentialAlert(message) ?
+      Promise.resolve(true) :
+      isCmasEnabledForServiceId(message.serviceId).catch((e) => {
+        console.error('CMAS: Unable to query settings database', e);
+        return false;
+      });
```

Comment:
I tried to integrate the previous check in this ternary but it was just making things less readable; so in the end I prefered to keep the first check separate.